Library/Homebrew/utils/tty.rb

```diff
@@ -7,7 +7,7 @@ def strip_ansi(string)
 
   def width
     width = `/bin/stty size 2>/dev/null`.split[1]
-    width ||= `/usr/bin/tput cols 2>/dev/null`.split[0]
+    width = `/usr/bin/tput cols 2>/dev/null`.split[0] if width.to_i.zero?
```

Comment:
Maybe it's worth handling the case where `tput cols` is also zero?

Reply:
Emmm, this?

```rb
  def width
    width = `/bin/stty size 2>/dev/null`.split[1]
    width = `/usr/bin/tput cols 2>/dev/null`.split[0] if width.to_i.zero?
    width = 80 if width.to_i.zero?
    width.to_i
  end
```
The code is ugly, and afaik, `tput cols 2>/dev/null ` returns "80\n" instead of zero in eshell without `COLUMNS` in environmental variables. Maybe it will never be zero.

Reply:
Yes, something like that seems superior. Could always make it a `while` loop or something?

Reply:
I don't know how to do it, my ruby-fu is poor...